workflow/state-machines.rst

```diff
@@ -0,0 +1,170 @@
+
+                        <framework:to>travis</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="update">
+                        <framework:from>coding</framework:from>
+                        <framework:from>travis</framework:from>
+                        <framework:from>review</framework:from>
+
+                        <framework:to>travis</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="wait_for_review">
+                        <framework:from>travis</framework:from>
+
+                        <framework:to>review</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="change_needed">
+                        <framework:from>review</framework:from>
+
+                        <framework:to>coding</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="accepted">
+                        <framework:from>review</framework:from>
+
+                        <framework:to>merged</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="rejected">
+                        <framework:from>review</framework:from>
+
+                        <framework:to>closed</framework:to>
+                    </framework:transition>
+
+                    <framework:transition name="reopened">
+                        <framework:from>closed</framework:from>
+
+                        <framework:to>review</framework:to>
+                    </framework:transition>
+
+                </framework:workflow>
+
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        use Symfony\Component\Workflow\Definition;
+        use Symfony\Component\Workflow\Transition;
+        use Symfony\Component\Workflow\StateMachine;
+        use Symfony\Component\Workflow\MarkingStore\SingleStateMarkingStore;
+
+        $states = ['start', 'coding', 'travis', 'review', 'merged', 'closed'];
+        $transitions[] = new Transition('submit', 'start', 'travis');
+        $transitions[] = new Transition('update', 'coding', 'travis');
+        $transitions[] = new Transition('update', 'travis', 'travis');
+        $transitions[] = new Transition('update', 'review', 'travis');
+        $transitions[] = new Transition('wait_for_reivew', 'travis', 'review');
+        $transitions[] = new Transition('change_needed', 'review', 'coding');
+        $transitions[] = new Transition('accepted', 'review', 'merged');
+        $transitions[] = new Transition('rejected', 'review', 'closed');
+        $transitions[] = new Transition('reopened', 'closed', 'review');
+
+        $definition = new Definition($states, $transitions, 'start');
+
+        $marking = new SingleStateMarkingStore('marking');
+        $stateMachine = new StateMachine($definition, $marking);
```

Comment:
Actually, we need to show how to configure the DI extension config instead of initially building the workflow itself (same below).